File: performance-investigation/fib/fib.ml
let rec fibonacci n =
  if n <= 1 then n
  else fibonacci (n - 1) + fibonacci (n - 2)

let () =
  match Array.length Sys.argv with
  | 2 ->
      let n = int_of_string Sys.argv.(1) in
      if n < 0 then
        print_endline "n must be a non-negative integer."
      else
        let result = fibonacci n in
        Printf.printf "The %dth number in the Fibonacci sequence is: %d\n" n result
  | _ ->
      Printf.printf "Usage: %s <n>\n" Sys.argv.(0)

File: performance-investigation/fib/fib.py
import sys

def fibonacci(n):
    if n <= 1:
        return n
    return fibonacci(n - 1) + fibonacci(n - 2)

if len(sys.argv) != 2:
    print("Usage: python fibonacci.py <n>")
    sys.exit(1)

n = int(sys.argv[1])
if n < 0:
    print("n must be a non-negative integer.")
    sys.exit(1)

result = fibonacci(n)
print(f"The {n}th number in the Fibonacci sequence is: {result}")